### ISSUE TITLE
[Feature suggestion] oauth login

### ISSUE THREAD

**Issue by parkerlreed** [michael-lazar/rtv#62]
Since reddit has moved to oauth it would be great to have that here. (Also makes me feel better than having to give my password directly to the application)

**Comment by michael-lazar** [comment]
I'm not too familiar with oauth but how would this work with an open source project? I can't force everybody to go to reddit and sign up for a developer key.

**Comment by Brobin** [comment]
I haven't touched Oauth that much in Python, but here is an open source project, [gcalcli](https://github.com/insanum/gcalcli/) that uses [Google Oauth](https://github.com/insanum/gcalcli/blob/master/gcalcli#L631).

**Comment by michael-lazar** [comment]
Looks like they slap it right in the code...
```
__API_CLIENT_ID__ = '232867676714.apps.googleusercontent.com'
__API_CLIENT_SECRET__ = '3tZSxItw6_VnZMezQwC8lUqy'
```

**Comment by Brobin** [comment]
Yeah...that's not how you're supposed to do it.
Everyone on [stackoverflow](http://stackoverflow.com/questions/1983990/how-to-open-source-an-application-that-uses-api-keys) suggests using a config file, which would require each user to supply their own keys.

**Comment by parkerlreed** [comment]
I was thinking like how newer mobile clients do it. They just drop to a webview to login to reddit. That way you can manage application access from within reddit.

![lxmog0z](https://cloud.githubusercontent.com/assets/841440/6982988/85c78810-d9e6-11e4-89a8-5bc94cdd8bdd.png)

**Comment by michael-lazar** [comment]
Reddit is deprecating cookie-based authentication on August 3rd, so this is now a huge priority. Even unauthenticated users will be throttled.

https://www.reddit.com/r/redditdev/comments/2ujhkr/important_api_licensing_terms_clarified/

**Comment by parkerlreed** [comment]
Thanks for the status update.

**Comment by shaggytwodope** [comment]
@michael-lazar so atm it wants the index.html file from ~/.config/rtv/templates/templates/index.html
It would be far more ideal and portable to have this read from /usr/share/rtv/templates/index.html or similar. So that it can be updated accordingly from various means of packaging. (same method used in setup.py to include manpage)

**Comment by michael-lazar** [comment]
@shaggytwodope what I ended up doing was moving the templates folder back inside of the project. That allowed me to configure setuptools to add the index.html to ``/usr/local/lib/python2.7/dist-packages/rtv/templates/``. I would have preferred to keep the `templates/` folder at the root of the project but I just couldn't get it to work without some giant workaround in setuptools.

**Comment by michael-lazar** [comment]
This is finally released as of v1.6!
